### PR TITLE
[FIX] lunch: make it work on mobile

### PR DESCRIPTION
**Pull request by pparidans**
This module didn't worked on mobile: user cannot select the products or
see their cart.

This commit reworks the LunchKanbanWidget template and introduces a
dedicated template for mobile. This template introduces a button at the
bottom of the screen to toggle the "cart" widget on mobile (hiding it by
default).

It tries to make the best use of the existing Bootstrap classes allowing
to have a more responsive layout and reducing the custom CSS required by
this module (both in desktop and in mobile).

This commit adds a decent JS tests-suite (mobile & desktop) to this
module as the existing one left substential parts of the UI untested
(opening wizard when clicking on a kanban record, cart's lines content,
clear cart button, widget structure and content based on the state...).
Those tests are also less based on DOM nodes order/tag names and use
instead dedicated classnames.

It also modifies the LunchKanbanRecord click handling to had a dedicated
event handler instead of overriding the default _onGlobalClick() and
make it more testable.

Finally this commit also remove unused code.

Task ID: 1945032

### CODE REVIEW

**Comment by pparidans** [comment]
Got the :+1: from EST. :sunglasses:

**Comment by pparidans** [comment]
As requested by @VincentSchippefilt a proper JS test-suite (at least I hope so :stuck_out_tongue_winking_eye: ) has been added...

@adr-odoo @VincentSchippefilt : let's fire :fire: up the reviews! I'm ready :man_firefighter:

File: addons/lunch/static/tests/lunch_kanban_mobile_tests.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+                    </templates>
+                </kanban>
+            `,
+            mockRPC: mockLunchRPC({
+                infos: this.regularInfos,
+                userLocation: this.data['lunch.location'].records[0].id,
+            }),
+        });
+
+        assert.containsN(kanban, '.o_kanban_view .o_kanban_record:not(.o_kanban_ghost)', 1,
```
Comment (VincentSchippefilt) [inline addons/lunch/static/tests/lunch_kanban_mobile_tests.js]: containsOnce

Comment (pparidans) [reply addons/lunch/static/tests/lunch_kanban_mobile_tests.js]: done ;)

**Comment by pparidans** [comment]
@robodoo retry

**Pull request closed without merging**